lib/psych/visitors/to_ruby.rb

```diff
@@ -12,6 +12,28 @@ module Visitors
     ###
     # This class walks a YAML AST, converting each node to Ruby
     class ToRuby < Psych::Visitors::Visitor
+      unless RUBY_VERSION < "3.2"
+        DATA_INITIALIZE = begin
+          init = Data.instance_method(:initialize)
+          if init == Object.instance_method(:initialize)
+            if RUBY_ENGINE == "truffleruby"
+              init = Object.new.tap do |o|
+                def o.bind_call(data, **members)
+                  data_mixin = data.class.ancestors
+                    .each_cons(2).find { Data.equal?(_2) }.first
+                  data_mixin
+                    .instance_method(:initialize)
+                    .bind_call(data, **members)
+                end
+              end.freeze
+            else
+              warn "Invalid Data.initialize"
+            end
+          end
+          init
+        end
```

Comment:
@eregon This is primarily meant as a starting point for discussion... and as something that passes all of the tests.  Is there a better way to identify TruffleRuby's specialized Data mixin?

Reply:
Yeah that's definitely too hacky :sweat_smile:

Reply:
@nevans Could you remove this workaround? Then I'll rerun the job once there is a new dev build of truffleruby-head.

Reply:
Dev build running in https://github.com/ruby/truffleruby-dev-builder/actions/runs/20148395191